lib/rubocop/cop/lint/void.rb

```diff
@@ -124,7 +124,8 @@ def check_void_expression(node)
         end
 
         def check_nonmutating(node)
-          return unless node.send_type? && NONMUTATING_METHODS.include?(node.method_name)
+          return unless (node.send_type? || node.block_type?) &&
```

Comment:
Can you add the numbered block parameter type?

```suggestion
          return unless (node.send_type? || node.block_type? || node.numblock_type?) &&
```